npm/private/npm_package.mjs

```diff
@@ -0,0 +1,14 @@
+import { spawnSync } from 'node:child_process'
+import fs from 'node:fs'
+
+const PACKAGE_DIR = `${process.env.JS_BINARY__RUNFILES}/{{PACKAGE_DIR}}`
```

Comment:
you could just pass PACKAGE_DIR as the first arg instead and avoid expand_template all together.. tho I suppose in that case the js_binary has to be run with `bazel run` and can't be run with the .sh script directly.

there is Bazel Slack thread going atm to add `fixed_args` to the js_binary public API so if that lands then we can use that here to bake the package dir arg into the js_binary

Reply:
https://github.com/aspect-build/rules_js/pull/1077

Reply:
Prerequisite of this pr; https://github.com/aspect-build/rules_js/pull/1079